config/static.c

```diff
@@ -246,9 +246,12 @@ config_gatekeeper(void)
 	int ret;
 	char lua_entry_path[128];
 	lua_State *lua_state;
+	const char *base = lua_base_dir == NULL ? LUA_BASE_DIR : lua_base_dir;
+	const char *file = gatekeeper_config_file == NULL ?
+		GATEKEEPER_CONFIG_FILE : gatekeeper_config_file;
 
 	ret = snprintf(lua_entry_path, sizeof(lua_entry_path), \
-			"%s/%s", LUA_BASE_DIR, GATEKEEPER_CONFIG_FILE);
+			"%s/%s", base, file);
 	RTE_VERIFY(ret > 0 && ret < (int)sizeof(lua_entry_path));
 
 	lua_state = luaL_newstate();
```

Comment:
I can't comment at line `set_lua_path(lua_state, LUA_BASE_DIR);` below, but you forgot to edit it.